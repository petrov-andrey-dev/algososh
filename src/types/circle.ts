import { ElementStates } from "./element-states";

export type TCircleObject = {
    value: string;
    state: ElementStates;
  }