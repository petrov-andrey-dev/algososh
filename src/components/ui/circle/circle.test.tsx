import renderer from 'react-test-renderer';

import { HEAD, TAIL } from '../../../constants/element-captions';
import { ElementStates } from '../../../types/element-states';
import { Circle } from './circle';

describe('Компонент Circle:', () => {
    it('Круг без буквы рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с буквами рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle letter='abcd'/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с head рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle head={HEAD} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с react-элементом в head рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle head={<Circle />} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с tail рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle tail={TAIL} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с react-элементом в tail рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle tail={<Circle />} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с index рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle index={0}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг с пропом isSmall ===  true рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle isSmall={true}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг в состоянии default  true рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Default}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг в состоянии changing  true рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Changing}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Круг в состоянии modified  true рендерится без ошибок', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Modified}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});