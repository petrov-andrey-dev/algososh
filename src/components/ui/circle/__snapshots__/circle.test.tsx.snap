// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Circle: Круг без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг в состоянии changing  true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг в состоянии default  true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг в состоянии modified  true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг с head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Компонент Circle: Круг с react-элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг с react-элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Компонент Circle: Круг с tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Компонент Circle: Круг с буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abcd
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle: Круг с пропом isSmall ===  true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
